### ISSUE TITLE
Show full column contents in SQL result

### ISSUE THREAD

**Issue by db2len** [IBM/db2forzosdeveloperextension-about#47]
Hi,

When executing an SQL statement that includes a CLOB column in the select list you are only shown the first 40-60 characters of the column's contents. It is very frustrating when you want to view the full contents of the column. 
There might be some reason behind this, but you ought to be able to decide in some way that you want to take the "risk" to see the full contents. Either by making it possible to extend the field where the column is displayed or by some setting. In other tools this is not a problem, some of them displays the full contents right away and in some you have the option to click on an ellipsis (...) to show the full contents. 

I've been trying to find some way to make this happen in Db2 for z/OS Developer Extension but have failed. Please advice if I've missed something otherwise I would like this function to be included as soon as possible.

Here is a generic example of an SQL statement against the Db2 for z/OS catalog:

```
select collid, name, version, statement
from sysibm.syspackstmt
where collid = 'mycoll';
```

The statement column is defined as CLOB(2097152) and contains static SQL statements.

Thanks in advance, Len

**Comment by realkendrickren** [comment]
@db2len Hi Len, thanks for bringing this up. 

The result is currently only showing first 50 characters for one column for performance sake, but I do agree that there are needs to see the entire content of a column for different reasons. We'll look into this request.

**Comment by db2len** [comment]
Yes, it would be very useful if you could chose to ask for the full column content. 

Many thanks for your willingness to look into this request!

/Len

**Comment by scotdb** [comment]
For both LOBs and (in our case mostly) XML columns, then it is good to show just the start of the column in the standard grid view.   But, just as in Data Studio, it would be great to have a little button appear when you hover over the column that you can then click to open the full column in another tab.   

In the case of XML columns, opening the new tab as a file suffixed .xml would then enable us to do many of the things we often have to do (using other tools readily available with VS Code).  Things like reformatting the XML to be more readable, checking it is valid and such like.

Maybe for BLOBs, which often hold multimedia data it might be useful to perhaps ask for a viewer at this stage too e.g. if the BLOB contains an image then suggest an image viewer.

**Comment by snewtonnj** [comment]
Any news on when this request will be implemented?  This is needed when selecting data from Db2 catalog tables that have CLOB columns.

**Comment by Thibaut69** [comment]
Same for me, a sub-view to watch the CLOB content would be greatly appreciated :)

**Comment by b-tsao** [comment]
Hi @db2len, in release 2.1.6 a feature that enable users to see an icon within the table view on hovering the CLOB allows user to open the CLOB and see the full contents in a new editor.
![image](https://github.com/user-attachments/assets/6159dc75-08ec-4a8d-88e3-b93a491f1fbb)
![image](https://github.com/user-attachments/assets/c7943c07-c03a-4f42-8637-0f88600dd669)

**Comment by amadhavia** [comment]
Delivered this in 2.1.6. Closing this enhancement request.
